webrtc/protocol/jsep-initial-offer.https.html

```diff
@@ -0,0 +1,30 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>RTCPeerConnection.prototype.createOffer</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../RTCPeerConnection-helper.js"></script>
+<script>
+  'use strict';
+
+  // Tests for the construction of initial offers according to
+  // draft-ietf-rtcweb-jsep-24 section 5.2.1
+  promise_test(async t => {
+    const pc = new RTCPeerConnection();
+    const offer = await pc.createOffer({offerToReceiveVideo: true});
+    let offer_lines = offer.sdp.split('\r\n');
+    // The first 3 lines are dictated by JSEP.
+    assert_equals(offer_lines[0], "v=0");
+    assert_equals(offer_lines[1].slice(0, 2), "o=");
+    // JSEP says that the address part SHOULD be a meaningless address
+    // "such as" IN IP4 127.0.0.1. We do strict matching here in order
+    // to detect if anyone ever uses something different.
+    assert_regexp_match(offer_lines[1], /^o=- \d+ \d+ IN IP4 127.0.0.1$/);
+    const fields = RegExp(/^o=- (\d+) (\d+)/).exec(offer_lines[1]);
```

Comment:
This fails on Firefox as it has the username field value "mozilla...THIS_IS_SDPARTA". JSEP says "`The value of the <username> field SHOULD be "-"`". Since it is optional it should not result in a test failure.

Reply:
@nils-ohlmeier says that Firefox expects to change this.
SHOULD is a problem for tests; if we don't test them, will they ever be tested?